sklearn/isotonic.py

```diff
@@ -252,7 +278,7 @@ def _build_f(self, X, y):
             # single y, constant prediction
             self.f_ = lambda x: y.repeat(x.shape)
         else:
-            self.f_ = interpolate.interp1d(X, y, kind='slinear',
+            self.f_ = interpolate.interp1d(X, y, kind='linear',
```

Comment:
The change on interp1d from linear to slinear was intended to solve many issues with Isotonic Regression https://github.com/scikit-learn/scikit-learn/issues/2507. I use Isotonic Regression in production code and indeed as soon as I upgraded to scikit-learn == 0.17.0 I saw a massive decrease in performance, slinear scales really bad with size, it can easily get to 1000 times slower. I wonder if changing it back to linear won't bring those issues back.